### PR TITLE
Fix PaymentMethod mapping 

### PR DESCRIPTION
**Pull request by nazli-stripe**
## Summary
Our current paymentMethod mapper maps the `paymentMethod.card` field instead of `paymentMethod.cardPresent`. This PR adds support for mapping card and interac present

## Motivation
https://github.com/stripe/stripe-terminal-react-native/issues/697

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: ios/Mappers.swift [hunk 1]
```diff
@@ -658,12 +658,19 @@ class Mappers {
...
 
     class func mapFromPaymentMethod(_ paymentMethod: PaymentMethod) -> NSDictionary {
+        var cardPresentMapped: NSDictionary?
+        if let cardPresent = paymentMethod.cardPresent{
+            cardPresentMapped = mapFromCardPresent(cardPresent)
+        }
+        var interacPresentMapped: NSDictionary?
+        if let interacPresent = paymentMethod.interacPresent{
+            interacPresentMapped = mapFromCardPresent(interacPresent)
+        }
```
Comment (bric-stripe) [inline ios/Mappers.swift:668]: thanks for fixing 🙏 

non-blocking suggestion: could also update mapFromCardPresent to accept a nullable `CardPresentDetails` and then can re-inline these.

File: ios/Mappers.swift [hunk 2]
```diff
@@ -658,12 +658,19 @@ class Mappers {
...
+
         let result: NSDictionary = [
-            "id": paymentMethod.stripeId,
-            "created": convertDateToUnixTimestamp(date: paymentMethod.created) ?? NSNull(),
-            "customer": paymentMethod.customer ?? NSNull(),
```
Comment (bric-stripe) [inline ios/Mappers.swift:664]: were these intentionally removed? 🤔

File: ios/Mappers.swift [hunk 3]
```diff
@@ -3,15 +3,15 @@ import StripeTerminal
 class Mappers {
     class func mapFromReaders(_ readers: [Reader]) -> [NSDictionary] {
         var readersList: [NSDictionary] = []
-
+
```
Comment (bric-stripe) [inline ios/Mappers.swift:6]: was this from Xcode re-indenting? for the native repo we run with the rm all whitespace options set and would be good to stay in sync there so we don't thrash 🤷‍♂️ 

in the "while editing" prefs:
Screenshot 2024-05-17 at 11 33 48

File: ios/Mappers.swift (hunk 2 above)
Comment (bric-stripe) [reply ios/Mappers.swift:664]: did you mean to rm `type` then and leave in `customer`?  looks like customer is there on both and metadata might be a good one to add while here too

> there is already a paymentMethodId field top level

the native SDKs duplicate this so may be worth it for consistency? I could see it being valuable if someone wanted to only pass around the PaymentMethod piece to work with 🤷‍♂️

**Comment by nazli-stripe** [comment]
@bric-stripe thanks for the review, addressed feedback

File: android/src/main/java/com/stripeterminalreactnative/Mappers.kt [hunk 4]
```diff
@@ -543,28 +559,30 @@ internal fun mapFromPaymentMethodDetailsType(type: PaymentMethodType?): String {
...
-        putString("iin", cardPresentDetails?.iin)
-        putString("network", cardPresentDetails?.network)
-        putString("description", cardPresentDetails?.description)
-        putMap("wallet", mapFromWallet(cardPresentDetails?.wallet))
-        putArray(
-            "preferredLocales",
-            convertListToReadableArray(cardPresentDetails?.preferredLocales)
-        )
+private fun mapFromCardPresentDetails(cardPresentDetails: CardPresentDetails?): ReadableMap? =
+    cardPresentDetails?.let {
```
Comment (nazli-stripe) [inline android/src/main/java/com/stripeterminalreactnative/Mappers.kt:563]: previously we were returning the map with all fields set to `null`, want to return just null instead

**Review by bric-stripe — APPROVED** [review]
lgtm, thanks for addressing the feedback

**Pull request merged**